terraform-modules/k8s/k8s-cluster.tf

```diff
@@ -3,49 +3,45 @@
 */
 
 resource "google_container_cluster" "cluster" {
-  provider            = "google"
-  name                = "${var.cluster_name}"
-  zone                = "${var.zone}"
+  provider = google-beta
+  name     = var.cluster_name
+  zone     = var.zone
 
-  network             = "${var.cluster_network}"
-  subnetwork          = "${var.cluster_subnetwork}"
+  network    = var.cluster_network
+  subnetwork = var.cluster_subnetwork
 
   # CIS compliance: stackdriver logging
-  logging_service     = "logging.googleapis.com"
+  logging_service = "logging.googleapis.com"
+
   # CIS compliance: stackdriver monitoring
-  monitoring_service  = "monitoring.googleapis.com"
+  monitoring_service = "monitoring.googleapis.com"
 
   # Defines the minimum version allowed for the K8s master
-  min_master_version  = "${var.master_version}"
-
-  # CIS compliance: disable legacy Auth
-  enable_legacy_abac  = "false"
+  min_master_version = var.master_version
 
   lifecycle {
-    ignore_changes  = [
-      "node_pool",
-      # this kept triggering a rebuild. No idea why
-      "master_auth.0.client_certificate_config.0.issue_client_certificate",
-
-      # It tries to update these to the specific format it likes even when the existing
-      # id is equivalent.
-      "network",
-      "subnetwork"
+    ignore_changes = [
+      node_pool,
+      master_auth[0].client_certificate_config[0].issue_client_certificate,
+      network,
+      subnetwork,
     ]
   }
 
   # Silly, but necessary to have a default pool of 0 nodes. This allows the node definition to be handled cleanly
   # in a separate file
-  node_pool {
-    name            = "default-pool"
-  }
+  remove_default_node_pool = true
```

Comment:
@RLuckom was there a reason why the existing code didn't use this argument? The TF examples recommend it as the way to get rid of the default node pool.

Reply:
there was no good reason. Either it didn't exist or I missed it